src/Indice.Features.Messages.AspNetCore/Endpoints/ContactsHandlers.cs

```
@@ -92,7 +92,32 @@ Guid contactId
         return TypedResults.Ok(preferences);
     }
 
+    public static async Task<Results<Ok<List<Contact?>>, NotFound>> GetDuplicateContacts(IContactService contactService, Guid contactId) {
+        Contact? contact = await contactService.GetById(contactId);
+        if (contact is null || string.IsNullOrWhiteSpace(contact.RecipientId)) {
+            return TypedResults.NotFound();
+        }
+        List<Contact?> duplicateContacts = await contactService.GetDuplicates(contact.RecipientId, contact.Email, contactId);
+        return TypedResults.Ok(duplicateContacts);
+    }
 
+    public static async Task<Results<Ok, ValidationProblem>> MergeContacts(IContactService contactService, Guid contactId, List<Guid> duplicateContactsIds) {
+        Contact? mainContact = await contactService.GetById(contactId);
+        if (mainContact is null) {
+            return TypedResults.ValidationProblem(ValidationErrors.AddError(nameof(mainContact), "Contact Id does not Exist"));
+        }
+        if(string.IsNullOrWhiteSpace(mainContact.RecipientId)) {
+            return TypedResults.ValidationProblem(ValidationErrors.AddError("Invalid Main Contact","The main contact does not have a recipient Id"));
+        }
+        if (duplicateContactsIds is null || duplicateContactsIds.Count == 0) {
+            return TypedResults.ValidationProblem(ValidationErrors.AddError(nameof(duplicateContactsIds), "Duplicates list cannot be empty"));
+        }
+        if (duplicateContactsIds.Contains(contactId)) {
```

It would be better to throw an error